sycl/test-e2e/Matrix/common.hpp

```diff
@@ -128,11 +137,12 @@ void matrix_copy(unsigned int rows, unsigned int cols, T *src, T *dst) {
   }
 }
 
-template <typename T1, typename T2>
+template <typename T1, typename T2, bool exact = false>
 bool matrix_compare(unsigned int rows, unsigned int cols, T1 *src, T2 *ref) {
   for (int i = 0; i < rows; i++) {
     for (int j = 0; j < cols; j++) {
-      if constexpr (std::is_same_v<T1, float> || std::is_same_v<T1, bfloat16>) {
+      if constexpr ((!exact) && std::is_same_v<T1, float> ||
+                    std::is_same_v<T1, bfloat16>) {
```

Comment:
I think condition should be this one, so that exact parameter is provided, type is not important anymore:

```suggestion
      if constexpr (!exact && (std::is_same_v<T1, float> ||
                    std::is_same_v<T1, bfloat16>)) {
```